netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -328,6 +352,36 @@ public void run() {
           keepAliveTimeInNanos, keepAliveTimeoutInNanos, true /* keepAliveDuringTransportIdle */);
       keepAliveManager.onTransportStarted();
     }
+    if (transportTracer != null) {
+      class FlowControlPollable implements Callable<Integer> {
+        private final Http2FlowController controller;
+
+        private FlowControlPollable(Http2FlowController controller) {
+          this.controller = controller;
+        }
+
+        @Override
+        public Integer call() throws Exception {
+          if (ctx.executor().inEventLoop()) {
+            return controller.windowSize(connection.connectionStream());
+          } else {
+            final Promise<Integer> promise = ctx.executor().newPromise();
+            ctx.executor().submit(new Runnable() {
+              @Override
+              public void run() {
+                promise.setSuccess(controller.windowSize(connection.connectionStream()));
+              }
+            });
+            return promise.get();
+          }
+        }
+      }
+
+      transportTracer.setRemoteFlowControlWindowPollable(
```

Comment:
Since these are often called together, it would probably be a good idea to bundle the results and batch the calls.

Reply:
done